packages/backend/src/outgoing_payment/lifecycle.ts

```diff
@@ -0,0 +1,341 @@
+
+export type PaymentError = LifecycleError | Pay.PaymentError
+export enum LifecycleError {
+  QuoteExpired = 'QuoteExpired',
+  // Rate fetch failed.
+  PricesUnavailable = 'PricesUnavailable',
+  // Payment aborted via "cancel payment" API call.
+  CancelledByAPI = 'CancelledByAPI',
+  // Not enough money in the super-account.
+  InsufficientBalance = 'InsufficientBalance',
+  // Error from the account service, except an InsufficientBalance. (see: CreditError)
+  AccountServiceError = 'AccountServiceError',
+
+  // These errors shouldn't ever trigger (impossible states), but they exist to satisfy types:
+  MissingQuote = 'MissingQuote',
+  InvalidRatio = 'InvalidRatio'
+}
+
+export async function handleQuoting(
+  deps: ServiceDependencies,
+  payment: OutgoingPayment,
+  plugin: IlpPlugin
+): Promise<void> {
+  const prices = await deps.ratesService.prices().catch((_err: Error) => {
+    throw LifecycleError.PricesUnavailable
+  })
+
+  const destination = await Pay.setupPayment({
+    plugin,
+    paymentPointer: payment.intent.paymentPointer,
+    invoiceUrl: payment.intent.invoiceUrl
+  })
+
+  assert.equal(
+    destination.destinationAsset.scale,
+    payment.destinationAccount.scale,
+    'destination scale mismatch'
+  )
+  assert.equal(
+    destination.destinationAsset.code,
+    payment.destinationAccount.code,
+    'destination code mismatch'
+  )
+
+  const quote = await Pay.startQuote({
+    plugin,
+    destination,
+    sourceAsset: {
+      scale: payment.sourceAccount.scale,
+      code: payment.sourceAccount.code
+    },
+    // This is always the full payment amount, even when part of that amount has already successfully been delivered.
+    // The quote's amounts are adjusted `handleSending` to reflect the actual payment state.
+    amountToSend: payment.intent.amountToSend,
+    slippage: deps.slippage,
+    prices
+  }).finally(() => {
+    return Pay.closeConnection(plugin, destination).catch((err) => {
+      deps.logger.warn(
+        {
+          destination: destination.destinationAddress,
+          error: err.message
+        },
+        'close quote connection failed'
+      )
+    })
+  })
+
+  await payment.$query(deps.knex).patch({
+    state: PaymentState.Ready,
+    quote: {
+      timestamp: new Date(),
+      activationDeadline: new Date(Date.now() + deps.quoteLifespan),
+      targetType: quote.paymentType,
+      minDeliveryAmount: quote.minDeliveryAmount,
+      maxSourceAmount: quote.maxSourceAmount,
+      // Cap at MAX_INT64 because of postgres type limits.
+      maxPacketAmount:
+        MAX_INT64 < quote.maxPacketAmount ? MAX_INT64 : quote.maxPacketAmount,
+      minExchangeRate: quote.minExchangeRate.valueOf(),
+      lowExchangeRateEstimate: quote.lowEstimatedExchangeRate.valueOf(),
+      highExchangeRateEstimate: quote.highEstimatedExchangeRate.valueOf()
+      //estimatedDuration: quote.estimatedDuration,
+    }
+  })
+}
+
+export async function handleReady(
```

Comment:
Because of how this (and the other `handle[state transition]` functions) are called, they are locked & in the expected state. I could add assertions for the state, though, if you think that would be clearer.

These functions aren't part of the service's public API, they are only ever called from `worker.ts`.